packages/steps/src/BuildStepContext.ts

```diff
@@ -233,9 +233,15 @@ export class BuildStepContext {
   }
 
   public get workingDirectory(): string {
-    return this.relativeWorkingDirectory
-      ? path.resolve(this.ctx.defaultWorkingDirectory, this.relativeWorkingDirectory)
-      : this.ctx.defaultWorkingDirectory;
+    if (!this.relativeWorkingDirectory) {
+      return this.ctx.defaultWorkingDirectory;
+    }
+
+    if (path.isAbsolute(this.relativeWorkingDirectory)) {
```

Comment:
nit: maybe we should rename `relativeWorkingDirectory` to just `workingDirectoryInput` or something like this since it might not be relative

Reply:
Yea, I wish all `steps` were significantly simpler. When we move `worker` in we may think of gradually simplifying all this.